server/services/reminderService.js

```diff
@@ -48,14 +67,29 @@ module.exports = function createReminderService(prisonerService, deadlineService
     }, Promise.resolve())
   }
 
-  async function getPrisonerData(bookingId, token) {
+  /**
+   * @typedef {[ResponsibleOfficerAndContactDetails, string]} NotificationData
+   *
+   * @param {number} bookingId
+   * @param {string} token
+   * @returns {Promise<Result<NotificationData>>}
+   */
+  async function getNotificationData(bookingId, token) {
     return Promise.all([
-      prisonerService.getOrganisationContactDetails('RO', bookingId, token),
-      prisonerService.getPrisonerPersonalDetails(bookingId, token),
-    ]).catch(error => {
-      logger.error('Error getting prisoner details for notification', error.stack)
-      return []
-    })
+      prisonerService.getEstablishmentForPrisoner(bookingId, token),
+      roContactDetailsService.getResponsibleOfficerWithContactDetails(bookingId, token),
+    ])
+      .then(([{ premise: prison }, roResult]) => raise(roResult, prison))
+      .then(result => {
+        if (isError(result)) {
+          logger.error('Error loading data for reminder', result)
+        }
+        return result
+      })
+      .catch(error => {
+        logger.error('Error loading data for reminder', error.stack)
+        return { message: `Error loading data for reminder: ${error.message}` }
+      })
   }
```

Comment:
If you're not going to use await then you don't need async.
Alternatively, replace the '.then' chain with await + try/catch

Reply:
I've replaced the `then` and `catch` with `await` and its lots clearer. Also meant I could get rid of the quite poor, additional methods in functionalHelpers